javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -266,13 +266,16 @@ Object getLeaf(int index) {
         } else if (depthShift == BRANCHING_BASE) {
             return obj().getAt(array, firstDigit(offset + index, depthShift));
         } else {
-            index += offset;
-            Object leaf = obj().getAt(array, firstDigit(index, depthShift));
-            for (int shift = depthShift - BRANCHING_BASE; shift > 0; shift -= BRANCHING_BASE) {
-                leaf = obj().getAt(leaf, digit(index, shift));
-            }
-            return leaf;
+            return getLeafGeneral(index);
+        }
+    }
+    private Object getLeafGeneral(int index) {
+        index += offset;
+        Object leaf = obj().getAt(array, firstDigit(index, depthShift));
+        for (int shift = depthShift - BRANCHING_BASE; shift > 0; shift -= BRANCHING_BASE) {
+            leaf = obj().getAt(leaf, digit(index, shift));
```

Comment:
Why don't we store the result of obj() once in a final local field? (=> only in several other locations of this PR)